components/patient-data/rest/src/main/java/org/phenotips/data/rest/internal/DefaultPatientsResourceImpl.java

```diff
@@ -93,34 +95,95 @@
-    {
+    public final Response add(final String json) {
         this.logger.debug("Importing new patient from JSON via REST: {}", json);
 
-        User currentUser = this.users.getCurrentUser();
+        final User currentUser = this.users.getCurrentUser();
         if (!this.access.hasAccess(Right.EDIT, currentUser == null ? null : currentUser.getProfileDocument(),
             this.currentResolver.resolve(Patient.DEFAULT_DATA_SPACE, EntityType.SPACE))) {
             throw new WebApplicationException(Status.UNAUTHORIZED);
         }
+
         try {
-            Patient patient = this.repository.create();
+            final Response response;
             if (json != null) {
-                try {
-                    patient.updateFromJSON(new JSONObject(json));
-                } catch (Exception ex) {
-                    throw new WebApplicationException(Status.BAD_REQUEST);
-                }
+                response = json.startsWith("[") ? addPatients(json) : addPatient(json);
+            } else {
+                response = buildResponse(this.repository.create());
             }
-            URI targetURI =
-                UriBuilder.fromUri(this.uriInfo.getBaseUri()).path(PatientResource.class).build(patient.getId());
-            ResponseBuilder response = Response.created(targetURI);
-            return response.build();
+            return response;
         } catch (Exception ex) {
             this.logger.error("Could not process patient creation request: {}", ex.getMessage(), ex);
             return Response.status(Status.INTERNAL_SERVER_ERROR).build();
         }
     }
 
+    /**
+     * Import new patients from their representation as a JSON array.
+     *
+     * @param json the JSON representation of the new patients to be created
+     * @return the location of the page displaying all patient data
+     * @throws WebApplicationException if a {@link JSONArray} object cannot be created or one of the patient objects
+     * is null
+     * @throws NullPointerException if the patient was not created
+     */
+    private Response addPatients(final String json) {
+        final JSONArray jsonArray;
+        try {
+            jsonArray = new JSONArray(json);
+        } catch (JSONException ex) {
+            throw new WebApplicationException(Status.BAD_REQUEST);
+        }
+
+        final int jsonArrayLength = jsonArray.length();
+        for (int i = 0; i < jsonArrayLength; i++) {
+            JSONObject jsonObject = jsonArray.optJSONObject(i);
+            if (jsonObject == null) {
+                logger.warn("One of the members of the patient JSONArray is null.");
+                continue;
+            }
+            Patient patient = this.repository.create();
+            patient.updateFromJSON(jsonObject);
+        }
+        final URI targetURI = UriBuilder.fromUri(this.uriInfo.getBaseUri()).build();
+        final ResponseBuilder response = Response.created(targetURI);
```

Comment:
This is not semantically correct, the `Location` header should point to the location of the created resource, not to a different resource. Since multiple resources are created, it would be better to use `null` for the header, and instead put the list of locations in the response body.